docker_images/diffusers/app/pipelines/image_to_image.py

```diff
@@ -0,0 +1,103 @@
+import json
+import os
+
+import torch
+from app.pipelines import Pipeline
+from diffusers import (
+    AltDiffusionImg2ImgPipeline,
+    ControlNetModel,
+    DiffusionPipeline,
+    DPMSolverMultistepScheduler,
+    StableDiffusionControlNetPipeline,
+    StableDiffusionImg2ImgPipeline,
+)
+from huggingface_hub import hf_hub_download, model_info
+from PIL import Image
+
+
+class ImageToImagePipeline(Pipeline):
+    def __init__(self, model_id: str):
+        model_data = model_info(model_id, token=os.getenv("HF_API_TOKEN"))
+
+        kwargs = (
+            {"safety_checker": None}
+            if model_id.startswith("hf-internal-testing/")
+            else {}
+        )
+        if torch.cuda.is_available():
+            kwargs["torch_dtype"] = torch.float16
+
+        has_config = any(
+            file.rfilename == "config.json" for file in model_data.siblings
+        )
+        if has_config:
+            config_file = hf_hub_download(
+                model_id, "config.json", token=os.getenv("HF_API_TOKEN")
+            )
+            with open(config_file, "r") as f:
+                config_dict = json.load(f)
+
+            is_controlnet = config_dict.get("_class_name", None) == "ControlNetModel"
+
+        if is_controlnet:
+            model_to_load = model_data.cardData["base_model"]
+            controlnet = ControlNetModel.from_pretrained(
+                model_id, use_auth_token=os.getenv("HF_API_TOKEN"), **kwargs
+            )
+
+            self.ldm = StableDiffusionControlNetPipeline.from_pretrained(
+                model_to_load,
+                controlnet=controlnet,
+                use_auth_token=os.getenv("HF_API_TOKEN"),
+                **kwargs,
+            )
+        else:
+            self.ldm = DiffusionPipeline.from_pretrained(
```

Comment:
This can load both `StableDiffusionImg2ImgPipeline` and `AltDiffusionImg2ImgPipeline` depending on how the model_index.json is defined